dashboard/src/app/actions/geography.ts

```diff
@@ -35,11 +35,19 @@ export const getWorldMapDataAlpha2 = withDashboardAuthContext(
       throw new Error(`Invalid parameters: ${validatedParams.error.message}`);
     }
 
-    const { startDate, endDate, queryFilters } = validatedParams.data;
+    const { startDate, endDate, queryFilters, compareStartDate, compareEndDate } = validatedParams.data;
 
     try {
       const geoVisitors = await fetchVisitorsByGeography(ctx.siteId, startDate, endDate, queryFilters);
-      return worldMapResponseSchema.parse(dataToWorldMap(geoVisitors, CountryCodeFormat.Original));
+
+      const compareGeoVisitors =
+        compareStartDate &&
+        compareEndDate &&
+        (await fetchVisitorsByGeography(ctx.siteId, compareStartDate, compareEndDate, queryFilters));
+
+      return worldMapResponseSchema.parse(
+        dataToWorldMap(geoVisitors, compareGeoVisitors ?? [], CountryCodeFormat.Original),
+      );
```

Comment:
```suggestion
      return dataToWorldMap(geoVisitors, compareGeoVisitors ?? [], CountryCodeFormat.Original);
```

The presenter layer already outputs the data as a WorldMapResponse, so there's no need to parse it here 😄

Reply:
Great, I removed the parse call and the schema import